adal/src/main/java/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -336,11 +336,14 @@ public void clearTokensForUser(String userId) {
 
         while (results.hasNext()) {
             final TokenCacheItem tokenCacheItem = results.next();
-                if (isAboutToExpire(tokenCacheItem.getExpiresOn())) {
+            if (tokenCacheItem.getExtendedExpiresOn() != null) {
+                if (isAboutToExpire(tokenCacheItem.getExtendedExpiresOn())) {
                     tokenItems.add(tokenCacheItem);
                 }
+            } else if (isAboutToExpire(tokenCacheItem.getExpiresOn())) {
+                tokenItems.add(tokenCacheItem);
```

Comment:
So you mean we do not do any deletion on cache items in Android, right? If so, you are right. This should not be involved with the ext_expires_in.